test/nlpmodels.jl

```diff
@@ -0,0 +1,92 @@
+using ADTypes: ADTypes
+using SparseConnectivityTracer
+import SparseConnectivityTracer as SCT
+using OptimizationProblems
+using ADNLPModels
+using NLPModels
+using NLPModelsJuMP
+using LinearAlgebra
+using SparseArrays
+using Test
+
+function mycons(nlp, x)
+    c = similar(x, nlp.meta.ncon)
+    cons!(nlp, x, c)
+    return c
+end
+
+function jac_sparsity_sct(name::Symbol)
+    nlp = OptimizationProblems.ADNLPProblems.eval(name)()
+    f(x) = mycons(nlp, x)
+    try
+        return ADTypes.jacobian_sparsity(f, nlp.meta.x0, TracerSparsityDetector())
+    catch e
+        @warn "Global Jacobian sparsity failed" name typeof(e)
+        return ADTypes.jacobian_sparsity(f, nlp.meta.x0, TracerLocalSparsityDetector())
+    end
```

Comment:
Can we not explicitly separate the problems into local and global?

Reply:
We could do that a posteriori based on which of these throw warnings, but there are over 200 and I don't want to make manual lists if I can avoid them